src/libmongoc/src/mongoc/mongoc-error.c

```diff
@@ -226,3 +224,84 @@ _mongoc_error_copy_labels_and_upsert (const bson_t *src,
 
    bson_append_array_end (dst, &dst_labels);
 }
+
+/* Defined in SDAM spec under "Application Errors".
+ * @error should have been obtained from a command reply, e.g. with
+ * _mongoc_cmd_check_ok.
+ */
+bool
+_mongoc_error_is_shutdown (bson_error_t *error)
+{
+   if (error->domain != MONGOC_ERROR_SERVER &&
+       error->domain != MONGOC_ERROR_WRITE_CONCERN) {
+      return false;
+   }
+   switch (error->code) {
+   case 11600: /* InterruptedAtShutdown */
```

Comment:
Nit: If these were an enum we could eliminate the comments entirely. We also wouldn't have to worry about grepping for numbers is this ever changes.

Reply:
Done. Also this made me realize https://jira.mongodb.org/browse/CDRIVER-3673